src/App.tsx

```diff
@@ -10,37 +11,28 @@ export const tabs = [
 ];
 
 export const App: React.FC = () => {
-  return (
-    <div className="section">
-      <h1 className="title">Selected tab is Tab 1</h1>
-
-      <div data-cy="TabsComponent">
-        <div className="tabs is-boxed">
-          <ul>
-            <li className="is-active" data-cy="Tab">
-              <a href="#tab-1" data-cy="TabLink">
-                Tab 1
-              </a>
-            </li>
+  const [selectedTabId, setSelectedTabId] = useState(tabs[0].id);
 
-            <li data-cy="Tab">
-              <a href="#tab-2" data-cy="TabLink">
-                Tab 2
-              </a>
-            </li>
+  const onTabSelected = (tab: {
+    id: string;
+    title: string;
+    content: string;
+  }) => {
+    setSelectedTabId(tab.id);
+  };
 
-            <li data-cy="Tab">
-              <a href="#tab-3" data-cy="TabLink">
-                Tab 3
-              </a>
-            </li>
-          </ul>
-        </div>
+  return (
+    <div className="section">
+      <h1 className="title">
+        Selected tab is{' '}
+        {tabs.filter(tab => tab.id === selectedTabId).map(tab => tab.title)}
```

Comment:
Let's create a variable for the  selected tab to avoid placing that logic into return function.